src/client/Client.js

```diff
@@ -471,6 +466,9 @@ class Client extends BaseClient {
     if (typeof options.internalSharding !== 'boolean') {
       throw new TypeError('CLIENT_INVALID_OPTION', 'internalSharding', 'a boolean');
     }
+    if (options.internalSharding && options.shardId && options.shardCount) {
```

Comment:
internal sharding should be detected by `shardCount: xyz` and no `shardId`

Reply:
I'm using the reference someone put in before hand which makes more sense, options.internalSharding. It also makes way more sense to use a boolean instead of shardCount because then it will use the recommended amount of shards and do what it's intended to do; handle everything itself.

Reply:
Also, this would cause an issue when using Shard Manager as for every client Shard Manager sets the shardCount and shardId variable. By detecting internal sharding with only shardCount, by using the Shard Manager (or custom sharding, if you'd prefer) would trigger Internal Sharding to start working causing a conflict.